### PR TITLE
[Hotfix] List more than 10 Folders through API [OSF-6830]

### PR DESCRIPTION
**Pull request by mfraezz**
## Purpose
List more than 10 folders for addons when using APIv2

## Changes
* Page size set to 1k
* Opportunistic Fix: `InstitutionsList` page size set to 1k:

>I do see us going over 100 institutions but we will likely remain under 1k
\- @icereval 

## Side effects
This isn't pagination. Because [where this is being called in TreeBeard](https://github.com/CenterForOpenScience/treebeard/blob/develop/dist/treebeard.js#L1064) doesn't resolve promises, this flow is difficult to hook into.

Consider adding a JSONAPI-compliant pagination feature to TreeBeard.

## Ticket
[OSF-6830](https://openscience.atlassian.net/browse/OSF-6830)

### CODE REVIEW

File: api_tests/base/test_pagination.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
+from nose.tools import *  # flake8: noqa
+
+from tests.base import ApiTestCase
+
+from api.base.pagination import MaxSizePagination
+
+class TestMaxPagination(ApiTestCase):
+    def test_no_query_param_alters_page_size(self):
+        assert_is_none(MaxSizePagination.page_size_query_param)
```
Comment (brianjgeiger) [inline api_tests/base/test_pagination.py]: Put an error message on this to say, "Adding variable page sizes to the paginator requires tests to ensure that you can't request more than the class's maximum number of values." So if they break it, they know what they need to do to fix it, and if it's removed in a PR, that the code reviewer will have a chance to see what it should have been replaced by.

Comment (brianjgeiger) [inline api_tests/base/test_pagination.py]: Or you can add in support for the page size and make sure that it works properly. Either way.

**Comment by brianjgeiger** [comment]
Looks good. 🎉 🐧 🎉

File: api/base/pagination.py [hunk 2]
```diff
@@ -128,6 +128,11 @@ def paginate_queryset(self, queryset, request, view=None):
             return super(JSONAPIPagination, self).paginate_queryset(queryset, request, view=None)
 
 
+class MaxSizePagination(JSONAPIPagination):
+    page_size = 1000
```
Comment (icereval) [inline api/base/pagination.py]: Can we not have just made the page_size get set to the length of the queryset results (or equivalent)? Thus not much of a max but a full results set on a single page...
